src/vulnerabilities/handler.js

```diff
@@ -0,0 +1,238 @@
+ */
+export const opportunityAndSuggestionsStep = async (auditUrl, auditData, context, site) => {
+  const { log, dataAccess } = context;
+  const { Configuration, Suggestion } = dataAccess;
+
+  // Check whether the audit is enabled for the site
+  const configuration = await Configuration.findLatest();
+  if (!configuration.isHandlerEnabledForSite('security-vulnerabilities', site)) {
+    log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] audit is disabled for site`);
+    return { status: 'complete' };
+  }
+
+  // This opportunity is only relevant for aem_cs delivery-type at the moment
+  if (site.getDeliveryType() !== 'aem_cs') {
+    log.debug(`[${AUDIT_TYPE}] [Site: ${site.getId()}] skipping vulnerability opportunity as it is of delivery type ${site.getDeliveryType()}`);
+    return { status: 'complete' };
+  }
+
+  const { vulnerabilityReport, success } = auditData.auditResult;
+
+  if (!success) {
+    log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] Audit failed, skipping opportunity / suggestions generation`);
+    return { status: 'complete' };
+  }
+
+  if (!isNonEmptyArray(vulnerabilityReport.vulnerableComponents)) {
+    // No vulnerabilities found
+    // Fetch opportunity
+    const { Opportunity } = dataAccess;
+    let opportunity;
+    try {
+      const opportunities = await Opportunity.allBySiteIdAndStatus(
+        site.getId(),
+        Oppty.STATUSES.NEW,
+      );
+      opportunity = opportunities.find((o) => o.getType() === AUDIT_TYPE);
+    } catch (e) {
+      log.error(`Fetching opportunities for siteId ${site.getId()} failed with error: ${e.message}`);
+      throw new Error(`Failed to fetch opportunities for siteId ${site.getId()}: ${e.message}`);
+    }
+
+    if (opportunity) {
+      // No vulnerabilities found, update opportunity status to RESOLVED
+      log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] no vulnerabilities found, but found opportunity, updating status to RESOLVED`);
+      await opportunity.setStatus(Oppty.STATUSES.RESOLVED);
+
+      // We also need to update all suggestions inside this opportunity
+      // Get all suggestions for this opportunity
+      const suggestions = await opportunity.getSuggestions();
+
+      // If there are suggestions, update their status to outdated
+      if (isNonEmptyArray(suggestions)) {
+        await Suggestion.bulkUpdateStatus(suggestions, SuggestionDataAccess.STATUSES.FIXED);
+      }
+      opportunity.setUpdatedBy('system');
+      await opportunity.save();
+    }
+
+    return { status: 'complete' };
+  }
+
+  // Update opportunity
+  const opportunity = await convertToOpportunity(
+    auditUrl,
+    { siteId: auditData.siteId, id: auditData.auditId },
+    context,
+    createOpportunityData,
+    AUDIT_TYPE,
+    createOpportunityProps(auditData.auditResult.vulnerabilityReport),
+  );
+
+  if (!configuration.isHandlerEnabledForSite('security-vulnerabilities-auto-suggest', site)) {
```

Comment:
the convertToOpportunity only creates an opportunity but without any libraries listed

If the flag is disabled, there will then be an empty opportunity with no suggestions?
Not sure we need a flag here - or if we have it, it should rather just set "empty" on the recommendedVersion received from the report?